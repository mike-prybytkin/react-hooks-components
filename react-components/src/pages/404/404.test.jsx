import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Page404 from './404';
import mockText from '../../mocks/text';

const setUp = () =>
  render(<Page404 heading={mockText.heading404} backToHomelinkText={mockText.linkBackToHome} />, {
    wrapper: BrowserRouter,
  });

describe('Page 404 component', () => {
  it('should correctly render heading', () => {
    setUp();
    const page404Heading = screen.getByText(new RegExp(mockText.heading404, 'i'));
    expect(page404Heading).toBeInTheDocument();
  });

  it('should correctly render link', () => {
    setUp();
    const linkHome = screen.getByText(new RegExp(mockText.linkBackToHome, 'i'));
    expect(linkHome).toBeInTheDocument();
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
