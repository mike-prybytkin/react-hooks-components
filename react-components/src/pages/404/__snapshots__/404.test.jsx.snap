// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page 404 component should create snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-container"
  >
    <h1>
      Error 404!
    </h1>
    <a
      class="back-to-home-link"
      href="/"
    >
      Back to home page
    </a>
  </div>
</DocumentFragment>
`;
